### PR TITLE
ES : move to functional tests

### PR DESCRIPTION
**Pull request by nplanel**

### CODE REVIEW

File: tests/storage_test.go [hunk 1]
```diff
@@ -81,6 +74,23 @@ func initBackend(t *testing.T, cfg elasticsearch.Config, name string) (*graph.El
...
+	return client, nil
+}
+
+func initBackend(t *testing.T, cfg elasticsearch.Config, name string) (*graph.ElasticSearchBackend, error) {
+	mappings := elasticsearch.Mappings{
+		{"node": []byte(graph.ESGraphElementMapping)},
+		{"edge": []byte(graph.ESGraphElementMapping)},
+	}
+
+	cfg.ElasticHost = "http://localhost:9200"
```
Comment (hunchback) [inline tests/storage_test.go]: No needed (default is 'http://127.0.0.1:9200')

File: tests/storage_test.go [hunk 2]
```diff
@@ -158,3 +168,180 @@ func TestElasticsearcActiveEdges(t *testing.T) {
...
+
+func indexEntry(c *elasticsearch.ElasticSearchClient, id int) (bool, error) {
+	return c.Index("test_type", fmt.Sprintf("id%d", id), `{"key": "val"}`)
+}
+
+// test rolling elasticsearch indices based on count limit
+func TestElasticsearchShouldRollByCount(t *testing.T) {
+	cfg := elasticsearch.NewConfig()
+	cfg.EntriesLimit = 5
+	name := "should_roll_by_count_test"
```
Comment (hunchback) [inline tests/storage_test.go]: t.Name() will return test name

File: tests/storage_test.go [hunk 3]
```diff
@@ -158,3 +168,180 @@ func TestElasticsearcActiveEdges(t *testing.T) {
...
+	if err := delTestIndex(name); err != nil {
+		t.Fatalf("Failed to clear test indices: %s", err.Error())
+	}
+}
+
+// test rolling elasticsearch indices based on age limit
+func TestElasticsearchShouldRollByAge(t *testing.T) {
+	cfg := elasticsearch.NewConfig()
+	cfg.AgeLimit = 5
+	name := "should_roll_by_age_test"
```
Comment (hunchback) [inline tests/storage_test.go]: t.Name()

File: tests/storage_test.go [hunk 4]
```diff
@@ -158,3 +168,180 @@ func TestElasticsearcActiveEdges(t *testing.T) {
...
+	if err := delTestIndex(name); err != nil {
+		t.Fatalf("Failed to clear test indices: %s", err.Error())
+	}
+}
+
+// test deletion of rolling elasticsearch indices
+func TestElasticsearchDelIndices(t *testing.T) {
+	cfg := elasticsearch.NewConfig()
+	cfg.IndicesLimit = 5
+	name := "del_indices_test"
```
Comment (hunchback) [inline tests/storage_test.go]: t.Name()

File: tests/storage_test.go [hunk 5]
```diff
@@ -158,3 +168,180 @@ func TestElasticsearcActiveEdges(t *testing.T) {
...
+	if err := delTestIndex(name); err != nil {
+		t.Fatalf("Failed to clear test indices: %s", err.Error())
+	}
+
+}
+
+// test mappings before and after rolling elasticsearch indices
+func TestElasticsearchMappings(t *testing.T) {
+	cfg := elasticsearch.NewConfig()
+	name := "mappings_test"
```
Comment (hunchback) [inline tests/storage_test.go]: t.Name()

File: tests/storage_test.go [hunk 6]
```diff
@@ -25,13 +25,15 @@ package tests
...
 	"net/http"
+	"strings"
 	"testing"
 	"time"
 
 	"github.com/skydive-project/skydive/storage/elasticsearch"
 	"github.com/skydive-project/skydive/topology/graph"
 )
 
+const indexPrefix = "test_skydive_mwnxf0f2"
```
Comment (hunchback) [inline tests/storage_test.go]: Maybe put the three following into a single const block, and remove redundent 'string' type specification.

Why is the 'mwnxf0f2' needed?

**Comment by nplanel** [comment]
close this PR as it's included in #1008

**Pull request closed without merging**